src/main/java/org/opensearch/security/dlic/rest/api/AuthFailureListenersApiAction.java

```diff
@@ -0,0 +1,265 @@
+                SecurityJsonNode authFailureListener = new SecurityJsonNode(body);
+                ValidationResult<SecurityJsonNode> validationResult = validateAuthFailureListener(authFailureListener, listenerName);
+
+                if (!validationResult.isValid()) {
+                    badRequest(channel, validationResult.toString());
+                    return;
+                }
+                String authenticationBackend = authFailureListener.get(TYPE_JSON_PROPERTY).asString().equals(IP_TYPE)
+                    || authFailureListener.get(AUTHENTICATION_BACKEND_JSON_PROPERTY).isNull()
+                        ? null
+                        : authFailureListener.get(AUTHENTICATION_BACKEND_JSON_PROPERTY).asString();
+                List<String> ignoreHosts = authFailureListener.get(IGNORE_HOSTS_JSON_PROPERTY).isNull()
+                    ? Collections.emptyList()
+                    : authFailureListener.get(IGNORE_HOSTS_JSON_PROPERTY).asList();
+
+                // Try to put the listener by name
+                config.dynamic.auth_failure_listeners.getListeners()
+                    .put(
+                        listenerName,
+                        new ConfigV7.AuthFailureListener(
+                            authFailureListener.get(TYPE_JSON_PROPERTY).asString(),
+                            Optional.ofNullable(authenticationBackend),
+                            ignoreHosts,
+                            authFailureListener.get(ALLOWED_TRIES_JSON_PROPERTY).asInt(),
+                            authFailureListener.get(TIME_WINDOW_SECONDS_JSON_PROPERTY).asInt(),
+                            authFailureListener.get(BLOCK_EXPIRY_JSON_PROPERTY).asInt(),
+                            authFailureListener.get(MAX_BLOCKED_CLIENTS_JSON_PROPERTY).asInt(),
+                            authFailureListener.get(MAX_TRACKED_CLIENTS_JSON_PROPERTY).asInt()
+                        )
+                    );
+                saveOrUpdateConfiguration(client, configuration, new OnSucessActionListener<>(channel) {
+                    @Override
+                    public void onResponse(IndexResponse indexResponse) {
+
+                        ok(channel, authFailureContent(config));
+                    }
+                });
+            }).error((status, toXContent) -> response(channel, status, toXContent)));
+
+    }
+
+    private ValidationResult<SecurityJsonNode> validateAuthFailureListener(SecurityJsonNode authFailureListener, String name) {
+        if (name == null) {
+            return ValidationResult.error(RestStatus.BAD_REQUEST, badRequestMessage("name is required"));
+        }
+        if (authFailureListener.get(TYPE_JSON_PROPERTY).isNull()) {
+            return ValidationResult.error(RestStatus.BAD_REQUEST, badRequestMessage("type is required"));
+        }
+        if (!(authFailureListener.get(TYPE_JSON_PROPERTY).asString().equals(IP_TYPE)
```

Comment:
```
 if (!(Set.of(IP_TYPE, USERNAME_TYPE).includes(authFailureListener.get(TYPE_JSON_PROPERTY).asString())) {
    ...
 }
```

Reply:
This is much cleaner! Fixed in: [2a8c237](https://github.com/opensearch-project/security/pull/4641/commits/2a8c237f25239167cf854bc50143f9f7f85a4af9)